addons/misra.py

```diff
@@ -456,15 +457,30 @@ def getArguments(ftok):
 
 
 def isalnum(c):
-    return (c >= '0' and c <= '9') or (c >= 'a' and c <= 'z') or (c >= 'A' and c <= 'Z')
+    return c in string.digits or c in string.ascii_letters
 
 
-def isHexDigit(c):
-    return (c >= '0' and c <= '9') or (c >= 'a' and c <= 'f') or (c >= 'A' and c <= 'F')
+def isHexEscapeSequence(symbols):
+    """Checks that given symbols are valid hex escape sequence.
+    Reference: n1570 6.4.4.4"""
+    if len(symbols) < 3 or symbols[0] != '\\' or symbols[1] != 'x':
```

Comment:
you could compress this condition I think:

    if len(symbols) < 3 or symbols[:2] != '\\x':